sbc_translation/models/translate_connector.py

```diff
@@ -0,0 +1,159 @@
+
+        text_type_id = 2 if letter.direction == 'Supporter To Beneficiary'\
+            else 1
+
+        vals = {
+            'src_lang_id': src_lang_id,
+            'aim_lang_id': aim_lang_id,
+            'title': self.letter_name,
+            'file': self.letter_name + '.pdf',
+            'codega': sponsor.ref,
+            'gender': sponsor.title.name,
+            'name': sponsor.name,
+            'firstname': sponsor.firstname,
+            'code': child.code,
+            'kid_name': child.name,
+            'kid_firstname': child.firstname,
+            'age': child_age,
+            'kid_gender': child.gender,
+            'createdat': self.current_time,
+            'updatedat': self.current_time,
+            'priority_id': 1,  # Not urgent, default
+            'text_type_id': text_type_id,
+        }
+        return self.upsert("text", vals)
+
+    def upsert_translation(self, text_id, letter):
+        """Push or update translation (db table) on local translate platform
+        """
+
+        vals = {
+            'file': self.letter_name + '.tif',
+            'text_id': text_id,
+            'createdat': self.current_time,
+            'updatedat': self.current_time,
+            'toDo_id': 0,
+            'letter_odoo_id': letter.id,
+        }
+        return self.upsert("translation", vals)
+
+    def upsert_translation_status(self, translation_id):
+        """Push or update translation_status (db table) on local translate
+        platform
+        """
+        to_translate = 1
+        vals = {
+            'translation_id': translation_id,
+            'status_id': to_translate,
+            'createdat': self.current_time,
+            'updatedat': self.current_time,
+        }
+        return self.upsert("translation_status", vals)
+
+    def get_language_id(self, language_iso_code):
+        """ Returns the language's id in MySQL that has  GP_Libel pointing
+         to the iso_code given (returns -1 if not found). """
+        res = self.selectOne(
+            "SELECT id FROM language WHERE GP_Libel LIKE '{}'"
+            .format(language_iso_code))
+        return res['id'] if res else -1
+
+    def get_translated_letters(self):
+        """ Returns a list for dictionaries with translation and filename
+        (sponsorship_id is in the file name...) in MySQL translation_test
+        database that has translation_status to 'Traduit" (id = 3)
+        (returns -1 if not found). """
+        res = self.selectAll(
+            "SELECT tr.id, tr.letter_odoo_id, tr.text, l.GP_libel target_lang\
+            FROM translation_status trs\
+            INNER JOIN translation tr\
+            ON trs.translation_id = tr.id\
+            INNER JOIN text txt\
+            ON tr.text_id = txt.id\
+            INNER JOIN language l\
+            ON txt.aim_lang_id = l.id\
+            WHERE tr.letter_odoo_id IS NOT NULL\
+            AND trs.status_id = 3"
+        )
+        return res if res else -1

```

Comment:
selectAll returns an empty list if nothing is found. That would be more helpful than -1, as you can use a loop outside the function without testing the list contains something.